arrow-json/src/reader/mod.rs

```diff
@@ -17,9 +17,13 @@
 
 //! JSON reader
 //!
-//! This JSON reader allows JSON line-delimited files to be read into the Arrow memory
-//! model. Records are loaded in batches and are then converted from row-based data to
-//! columnar data.
+//! This JSON reader allows JSON records to be read into the Arrow memory
+//! model. Records are loaded in batches and are then converted from the record-oriented
+//! representation to the columnar arrow data model.
+//!
+//! The reader is agnostic to whitespace, including `\n` and `\r`, and will ignore such characters.
+//! This allows parsing sequences of one or more arbitrarily formatted JSON values, including
+//! but not limited to newline-delimited JSON.
```

Comment:
A minor suggestion to improve the wording here

```suggestion
//! The reader ignores whitespace between JSON values, including `\n` and `\r`.
//! This allows parsing sequences of one or more arbitrarily formatted JSON values, including
//! but not limited to newline-delimited JSON.
```